packages/backend/src/routers/oauthRouter.ts

```diff
@@ -241,6 +253,52 @@ oauthRouter.post('/revoke', async (req, res) => {
     return res.status(200).send();
 });
 
+// Dynamic Client Registration endpoint (RFC 7591)
+oauthRouter.post('/register', async (req, res) => {
+    try {
+        const {
+            client_name,
+            redirect_uris,
+            grant_types = ['authorization_code', 'refresh_token'],
+            scope = 'read write',
```

Comment:
let's introduce a specific mcp scope and have it such that _only_ mcp scopes can be accessed via DCR (this endpoint) - since this allows anybody to register an oauth client. we should also make sure we have logging for this on the AUDIT endpoint. also open a ticket on the infra repo for rate limiting